File: affy/sdk/parser_examples/cpp/read_chp/read_chp.cpp

#include "FusionCHPData.h"
#include "FusionCHPLegacyData.h"
#include <iostream>

using namespace std;
using namespace affymetrix_fusion_io;

int main(int argc, char **argv)
{
	const char* fileName = argv[1];
	try
	{
		FusionCHPData *chp = FusionCHPDataReg::Read(fileName);
		if (chp == NULL)
		{
			cout << "Error reading the file." << endl;
			return 0;
		}
		FusionCHPLegacyData *legchp = FusionCHPLegacyData::FromBase(chp);
		if (legchp == NULL)
		{
			cout << "The example is for expression CHP files only" << endl;
			return 0;
		}
		if (legchp->GetHeader().GetAssayType() != FusionExpression)
		{
			cout << "The example is for expression only." << endl;
			return 0;
		}

		FusionExpressionProbeSetResults psResults;
		float sum = 0;
		int n = legchp->GetHeader().GetNumProbeSets();
		for (int i = 0; i < n; i++)
		{
			legchp->GetExpressionResults(i, psResults);
			sum += psResults.GetSignal();
		}
		float avg = sum / n;
		cout << "The average signal is: " << avg << endl;

		// Delete the chp object.
		delete legchp;
	}
	catch (...)
	{
		cout << "Error reading the file." << endl;
	}
	return 0;
}

File: affy/sdk/parser_examples/java/read_chp/src/read_chp/read_chp.java
import affymetrix.fusion.chp.FusionCHPData;
import affymetrix.fusion.chp.FusionCHPDataReg;
import affymetrix.fusion.chp.FusionCHPHeader;
import affymetrix.fusion.chp.FusionCHPLegacyData;
import affymetrix.fusion.chp.FusionExpressionProbeSetResults;

public class read_chp {

	public read_chp() {
	}

	/**
	 * @param args
	 *          the command line arguments
	 */
	public static void main(String[] args) {

		FusionCHPLegacyData.registerReader();
		String fileName = args[0];
		try {
			FusionCHPData chp = FusionCHPDataReg.read(fileName);
			if (chp == null) {
				System.out.println("Failed to read the file.");
				return;
			}
			FusionCHPLegacyData legchp = FusionCHPLegacyData.fromBase(chp);
			if (legchp == null) {
				System.out.println("The example is for expression CHP files only.");
				return;
			}
			int atype = legchp.getHeader().getAssayType();
			if (atype != FusionCHPHeader.EXPRESSION_ASSAY) {
				System.out.println("The example is for expression only.");
				return;
			}

			float sum = 0;
			int nump = 0;
			int n = legchp.getHeader().getNumProbeSets();
			FusionExpressionProbeSetResults psResults = new FusionExpressionProbeSetResults();
			for (int i = 0; i < n; i++) {
				legchp.getExpressionResults(i, psResults);
				sum += psResults.getSignal();
				if (psResults.getDetection().toShort() == FusionExpressionProbeSetResults.ABS_PRESENT_CALL) {
					++nump;
				}
			}
			float avg = sum / n;
			System.out.println("The average signal is " + avg);
			System.out.println("The #P is " + nump + " (" + (int)(100.0 * nump / n) + "%)");
		}
		catch (Exception e) {
		}
	}

}
